app/models/book.rb

```diff
@@ -1,3 +1,5 @@
 class Book < ApplicationRecord
+  has_many :rents, dependent: :destroy
```

Comment:
if your reference to the book (and user) in your rent schema are nullable, it doesn't makes sence that the relation from book (and user) is dependent destroy, just go for `dependent: :nullify` or make book/user not nullable in rent